### PR TITLE
Building libproj from source on Linux

### PR DESCRIPTION
**Pull request by awbsmith**
###### The Good:

- It'll build Proj4 from source allowing for newer Proj4 versions than apt-get

###### The Bad:

- No minimum version for the libproj dependency (if you already have an old libproj version it won't build a newer one)
- I've only done it for Linux
- Slower load time and pre-compiling whinges.  
- There's got to be a way to get the library name and path from the dependency entry

###### The Other

- had to rename the library name variable to not conflict with the dependency entry

### CODE REVIEW

**Comment by awbsmith** [comment]
Hmm, the Ubuntu [Universe](http://www.ubuntuupdates.org/package/core/wily/universe/base/libproj-dev) ppa has libproj 4.9.1 anyway.  There might have been an easier way...

File: deps/build.jl [hunk 1]
```diff
@@ -2,10 +2,45 @@ using BinDeps
...
 @linux_only begin
-    provides(AptGet, "libproj0", libproj, os = :Linux)
+
+	
+	# AptGet's LibProj is old
+	#provides(AptGet, "libproj0", libproj, os = :Linux)  
+
+	# so download a newer version
+	libsrc = "http://download.osgeo.org/proj"
+	libproj_ver = "4.9.2" # from source http://download.osgeo.org/proj/
```
Comment (nigels-com) [inline deps/build.jl]: It's nice to be using a specific version, not just whatever the latest release is.

**Comment by nigels-com** [comment]
Does it work on RedHat, CentOS or Fedora?

**Comment by BuzzVII** [comment]
the big question is does it work on the Amazon linux ami. we need all these
to build and package in prod.

**Comment by nigels-com** [comment]
It's more likely to work on Amazon Linux (or anywhere else) if it isn't relying on some specific version installed system-wide, right?

**Comment by awbsmith** [comment]
Lets find out!

File: deps/build.jl [hunk 2]
```diff
@@ -2,10 +2,45 @@ using BinDeps
...
+	libproj_name = "proj-$(libproj_ver).tar.gz"
+
+	# and build it from source
+	prefix   = joinpath(BinDeps.depsdir(libproj), "usr")
+	srcdir   = joinpath(BinDeps.depsdir(libproj), "src", "proj-$(libproj_ver)")
+	builddir = joinpath(BinDeps.depsdir(libproj), "builds", "libproj-$(libproj_ver)_version")
+	libdir   = joinpath(prefix, "lib")
+
+	provides(Sources, URI(joinpath(libsrc, libproj_name)), libproj, os = :Linux, installed_libpath=libdir)
```
Comment (c42f) [inline deps/build.jl]: We should now be able to get away with `:Linux` -> `:Unix`, and `@unix_only`

File: deps/build.jl [hunk 3]
```diff
@@ -2,10 +2,45 @@ using BinDeps
...
+
+	provides(BuildProcess,
+		(@build_steps begin
+			GetSources(libproj)
+			CreateDirectory(builddir)
+			@build_steps begin
+				ChangeDirectory(builddir)
+				FileRule(joinpath(libdir, "libproj.so"), 
+				@build_steps begin
```
Comment (c42f) [inline deps/build.jl]: Let's indent as

```julia
FileRule(joinpath(libdir, "libproj.so"),
         @build_steps begin
            ChangeDirectory(srcdir)
            `./configure --prefix=$(prefix)`
            `make`
            `make install` 
         end)
```

**Comment by c42f** [comment]
@awbsmith argh.  Apparently I managed to push this all to master, along with my cleanups when I meant to push my extra changes to a branch.

Never mind, I think it's all good...  let me know if things are not as they seem.

**Pull request merged**